mitiq/utils.py

```diff
@@ -0,0 +1,74 @@
+"""Utility functions."""
+
+import random
+
+import cirq
+
+
+def random_circuit(depth: int, **kwargs) -> cirq.Circuit:
+    """Returns a random single-qubit circuit with Pauli gates.
+
+    Parameters
+    ----------
+        depth: Number of gates in the circuit.
+
+    Keyword Args
+    ------------
+        seed: Sets seed for the random number generator to input `seed`.
+    """
+    if "seed" in kwargs.keys():
+        random.seed(kwargs.get("seed"))
+
+    qubit = cirq.GridQubit(0, 0)
+    gates = [cirq.ops.X, cirq.ops.Y, cirq.ops.Z]
+    circuit = cirq.Circuit(
+        [random.choice(gates).on(qubit) for _ in range(depth)]
+    )
+
+    return circuit
+
+
+def _equal(
```

Comment:
This is a neat function, well done. I thought this would be much harder to implement but Cirq has lots of useful functions that you used well.

Reply:
The qubit equality aspect of comparing circuits is kind of annoying, but thankfully there was a relatively straightforward workaround with transforming qubits. The only requirement is that the qubits must be sortable. I think this is always the case, but if it's not there's really no way at all to "line up" qubits from one circuit to the other.